### PR TITLE
Set up associations

### PR DESCRIPTION
**Pull request by Chaseknis**
- Add `classroom.rb` file
- Add `rental.rb` file
- Add `book.rb`
- Implement `many-to-many` relationship
- Implement `has-many`/`belongs-to`, 
- Update`student.rb` file
- Update `person.rb` file
- Update `README.md` file
- Run `linters` for errors

### CODE REVIEW

File: student.rb [hunk 1]
```diff
@@ -3,7 +3,7 @@ class Student < Person
   attr_accessor :classroom
 
   def initialize(age:, classroom:, name: 'Unknown', parent_permission: true)
-    super(age: age, name: name, parent_permission: parent_permission)
+    super(name: name, age: age, parent_permission: parent_permission)
     @classroom = classroom
   end
```
Comment (AmaduKamara) [inline student.rb]: - [x] This class is expected to have a classroom method that serves as the 'belongs-to' side (a student belongs to a classroom). Kindly add that 👍🏽

File: book.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  # Other methods of Book class
```
Comment (AmaduKamara) [inline book.rb]: - [x] For this class, a method for adding rentals is required so rentals can be added. Kindly add that 👍🏽

**Review by AmaduKamara — CHANGES_REQUESTED** [review]
# Status: Required Changes ♻️ ♻️ ♻️

Hi @Chaseknis,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

✅ You used Gitflow correctly 👍🏽 
✅ Linter checks passed 👍🏽

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: student.rb [hunk 3]
```diff
@@ -1,9 +1,10 @@
 require './person'
+
 class Student < Person
-  attr_accessor :classroom
+  attr_reader :classroom
 
   def initialize(age:, classroom:, name: 'Unknown', parent_permission: true)
-    super(age: age, name: name, parent_permission: parent_permission)
+    super(name: name, age: age, parent_permission: parent_permission)
```
Comment (tufoinnkuo10) [inline student.rb:7]: - [x] please like the previous reviewer noted there shoudl be method that serves as the 'belongs-to' side (a student belongs to a classroom)
eg 
```ruby
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Chaseknis 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Gitflow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ associations created well

Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

**Review by algerina — APPROVED** [review]
Hi  @Chaseknis  ,

Great job so far! 

May I highlight the following: 
- [x] Your PR is descriptive✔️ 
- [x] Well done implementing all the requirements✔️ 


## STATUS: APPROVED :muscle::clap:

Congratulations! :tada:

Your project is complete!  It's time to merge it :shipit:



 
 ## Optional suggestions

 - Please check the comment under the review.

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
Cheers and Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread and tag me @algerina if something is not 100% clear.

#### _**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**_
------------------------------------------------------------------------------------------------------------------------------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)
___
_If anything seems to be unclear to you, reach out and ask anything by commenting on this Pull Request and mention me as [Algerina](https://github.com/algerina), I will receive a notification and will respond._

[GitHub](https://github.com/algerina) | [LinkedIn](https://linkedin.com/in/amel-khiri) | [Twitter](https://twitter.com/AmelQahwadji)

_Reviewed by [Amel Khiri ](https://github.com/algerina)_

**Comment by Chaseknis** [comment]
thank you @algerina

**Pull request merged**